Add notice what to do if you do not get email (#5599)
jenkins-infra/jenkins.io: content/doc/developer/publishing/preparation.adoc

== Signup Required

> Jenkins plugins distributed via Jenkins project update sites need to be hosted in the https://github.com/jenkinsci[jenkinsci GitHub organization], so you will need a user account on GitHub. 

Reviewer: ```suggestion
Jenkins plugins distributed via Jenkins project update sites need to be hosted in the https://github.com/jenkinsci[jenkinsci GitHub organization], so you will need a user account on GitHub.
```

> Jenkins plugins distributed via Jenkins project update sites need to be hosted in the https://github.com/jenkinsci[jenkinsci GitHub organization], so you will need a user account on GitHub. 

Reviewer 2: ```suggestion
Jenkins plugins distributed via Jenkins project update sites need to be hosted in the link:https://github.com/jenkinsci[jenkinsci GitHub organization], so you will need a user account on GitHub. 
```
just to make sure the `link:` macro is added

> To actually release your plugin, you will need a https://accounts.jenkins.io[Jenkins community account] that will give you access to the https://issues.jenkins.io/[Jira issue tracker] and https://repo.jenkins-ci.org/[Maven repository]. 

Reviewer: ```suggestion
To actually release your plugin, you will need a https://accounts.jenkins.io[Jenkins community account] that will give you access to the https://issues.jenkins.io/[Jira issue tracker] and https://repo.jenkins-ci.org/[Maven repository].
```

> To actually release your plugin, you will need a https://accounts.jenkins.io[Jenkins community account] that will give you access to the https://issues.jenkins.io/[Jira issue tracker] and https://repo.jenkins-ci.org/[Maven repository]. 

Reviewer 2: ```suggestion
To actually release your plugin, you will need a link:https://accounts.jenkins.io[Jenkins community account] that will give you access to the link:https://issues.jenkins.io/[Jira issue tracker] and link:https://repo.jenkins-ci.org/[Maven repository]. 
```
same as above, just making sure the `link:` macro is used

> You will receive a welcome email if your Jenkins community account was created successfully. If you do not receive an email, open a ticket at https://github.com/jenkins-infra/helpdesk.

Reviewer: ```suggestion
You should receive a welcome email confirming your Jenkins community account was created successfully.
If you do not receive any email, please open a ticket on the [Jenkins Infrastructure Help Desk](https://github.com/jenkins-infra/helpdesk/issues).
```

> You will receive a welcome email if your Jenkins community account was created successfully. If you do not receive an email, open a ticket at https://github.com/jenkins-infra/helpdesk.

Reviewer 2: ```suggestion
You will receive a welcome email if your Jenkins community account was created successfully.
If you do not receive an email, open a ticket at link:https://github.com/jenkins-infra/helpdesk[the Jenkins helpdesk].
```
small item to make sure each sentence has its own line, making sure the `link:` macro is used
